zk/legacy_executor_verifier/legacy_executor_verifier.go

```diff
@@ -94,101 +116,116 @@ func NewLegacyExecutorVerifier(
 		stream:           stream,
 		witnessGenerator: witnessGenerator,
 		l1Syncer:         l1Syncer,
-		promises:         make([]*Promise[*VerifierResponse], 0),
+		promises:         make([]*Promise[*VerifierBundle], 0),
 		addedBatches:     make(map[uint64]struct{}),
 	}
 }
 
 // var counter = int32(0)
 
-func (v *LegacyExecutorVerifier) AddRequestUnsafe(ctx context.Context, tx kv.RwTx, request *VerifierRequest) (*Promise[*VerifierResponse], error) {
-	// if we have no executor config then just skip this step and treat everything as OK
-	if !v.HasExecutors() {
-		response := &VerifierResponse{
-			BatchNumber: request.BatchNumber,
-			Valid:       true,
-		}
-		return &Promise[*VerifierResponse]{
-			result: response,
-			err:    nil,
-		}, nil
-	}
+// Unsafe is not thread-safe so it MUST be invoked only from a single thread
+func (v *LegacyExecutorVerifier) AddRequestUnsafe(request *VerifierRequest, sequencerBatchSealTime time.Duration) *Promise[*VerifierBundle] {
+	// eager promise will do the work as soon as called in a goroutine, then we can retrieve the result later
+	promise := NewPromise[*VerifierBundle](func() (*VerifierBundle, error) {
+		var err error
+		var tx kv.Tx
+		var hermezDb *hermez_db.HermezDbReader
+		var blocks []uint64
+		// hermezDb := hermez_db.NewHermezDbReader(tx)
+		startTime := time.Now()
+		ctx := context.Background()
+
+		// mapmutation has some issue with us not having a quit channel on the context call to `Done` so
+		// here we're creating a cancelable context and just deferring the cancel
+		innerCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		// get the data stream bytes
+		for time.Since(startTime) < 2*sequencerBatchSealTime {
+			tx, err = v.db.BeginRo(innerCtx)
+			if err != nil {
+				return nil, err
+			}
 
-	hermezDb := hermez_db.NewHermezDbReader(tx)
+			hermezDb = hermez_db.NewHermezDbReader(tx)
+			blocks, err = hermezDb.GetL2BlockNosByBatch(request.BatchNumber)
+			if err != nil {
+				tx.Rollback()
+				return nil, err
+			}
 
-	// get the data stream bytes
-	blocks, err := hermezDb.GetL2BlockNosByBatch(request.BatchNumber)
-	if err != nil {
-		return nil, err
-	}
+			// we might not have blocks yet as the underlying stage loop might still be running and the tx hasn't been
+			// committed yet so just requeue the request
+			if len(blocks) > 0 {
+				break
+			}
 
-	// we might not have blocks yet as the underlying stage loop might still be running and the tx hasn't been
-	// committed yet so just requeue the request
-	if len(blocks) == 0 {
-		request.CheckCount++
-		return nil, nil
-	}
+			tx.Rollback()
+			time.Sleep(time.Second)
```

Comment:
maybe reduce this down so we can get comms with the executor a little faster?  100ms? maybe

Reply:
I thought so but then I realized that executor is quite slower than our sequencer (magnitudes slower) so in reallity committing faster to the executor will actually means committing a new promise that wait for the executor queue to empty.

Otherwise, we can create a flag for this one or, best case, adjust time time dynamically based on usage data. What do you think?

Reply:
Let's keep it where it is for now.  If it becomes a problem we can always a flag for this.